Sources/ItemList.php

```diff
@@ -402,82 +402,72 @@ protected function buildRows(): void
 			$cur_row = [];
 
 			foreach ($this->options['columns'] as $column_id => $column) {
-				$cur_data = [];
+				$data = $column['data'] ?? [];
+				$cur_data = ['value' => $data['value'] ?? ''];
 
 				// A value straight from the database?
-				if (isset($column['data']['db'])) {
-					$cur_data['value'] = $list_item[$column['data']['db']];
+				if (isset($data['db'])) {
+					$cur_data['value'] = $list_item[$data['db']];
 				}
 				// Take the value from the database and make it HTML safe.
-				elseif (isset($column['data']['db_htmlsafe'])) {
-					$cur_data['value'] = Utils::htmlspecialchars((string) $list_item[$column['data']['db_htmlsafe']]);
+				elseif (isset($data['db_htmlsafe'])) {
+					$cur_data['value'] = Utils::htmlspecialchars((string) $list_item[$data['db_htmlsafe']]);
 				}
-				// Using sprintf is probably the most readable way of injecting data.
-				elseif (isset($column['data']['sprintf'])) {
-					$params = [];
-
-					foreach ($column['data']['sprintf']['params'] as $sprintf_param => $htmlsafe) {
-						$params[] = $htmlsafe ? Utils::htmlspecialchars((string) $list_item[$sprintf_param]) : $list_item[$sprintf_param];
+				// These are probably the most readable way of injecting complex data.
+				elseif (isset($data['sprintf']) || isset($data['formatText']) || isset($data['getTxt'])) {
+					$params = $list_item;
+					$call = 'vsprintf';
+					$format = 'sprintf';
+
+					if (isset($data['getTxt'])) {
+						$call = 'SMF\Lang::getTxt';
+						$format = 'getTxt';
+					} elseif (isset($data['formatText'])) {
+						$call = 'SMF\Lang::formatText';
+						$format = 'formatText';
 					}
```

Comment:
```suggestion
					if (isset($data['getTxt'])) {
						$call = 'SMF\Lang::getTxt';
						$format = 'getTxt';
					} else {
						$call = 'SMF\Lang::formatText';
						$format = isset($data['formatText']) ? 'formatText' : 'sprintf';
					}
```

SMF\Lang::formatText() handles both MessageFormat and sprintf format, so it can be called for both format types.